### PR TITLE
Made flann dependency for features2d optional

### PR DESCRIPTION
**Pull request by jet47**
It will allow to build features2d even if flann module is not available.

### CODE REVIEW

File: modules/features2d/include/opencv2/features2d.hpp [hunk 1]
```diff
@@ -1099,6 +1103,7 @@ class CV_EXPORTS_W BFMatcher : public DescriptorMatcher
     bool crossCheck;
 };
 
+#ifdef HAVE_OPENCV_FLANN
```
Comment (alalek) [inline modules/features2d/include/opencv2/features2d.hpp]: `#if defined(HAVE_OPENCV_FLANN) || defined(CV_DOXYGEN)`

Comment (jet47) [reply modules/features2d/include/opencv2/features2d.hpp]: Fixed.

Comment (alalek) [reply modules/features2d/include/opencv2/features2d.hpp]:
```.diff
-#ifdef
+#if
```

Comment (jet47) [reply modules/features2d/include/opencv2/features2d.hpp]: Sorry, my mistake.

**Comment by mshabunin** [comment]
:+1:

**Pull request merged**